### PR TITLE
fix(rule_engine): don't enable a rule that references non-existent bridge

### PR DESCRIPTION
**Pull request by thalesmg**
Fixes <issue-or-jira-number>

### <samp>🤖 Generated by Copilot at 3da0f05</samp>

This pull request enhances the rule engine module to handle bridge-related rules more robustly and reliably. It adds a new feature to disable invalid rules that reference non-existent bridges, and improves the test coverage and flexibility of the rule engine and bridge modules. It affects the files `emqx_rule_engine.erl`, `emqx_bridge_v1_compatibility_layer_SUITE.erl`, `emqx_rule_engine_SUITE.erl`, and `emqx_bridge_v2_testlib.erl`.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl [hunk 1]
```diff
@@ -478,6 +477,13 @@ with_parsed_rule(Params = #{id := RuleId, sql := Sql, actions := Actions}, Creat
                 conditions => emqx_rule_sqlparser:select_where(Select)
                 %% -- calculated fields end
             },
+            InputEnable = maps:get(enable, Params, true),
+            Enable =
+                case validate_bridge_existence_in_actions(Rule0) of
+                    ok -> InputEnable;
+                    {error, _} -> false
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_engine.erl:484]: this is something user needs to fix.
maybe write an error level log here

File: apps/emqx_bridge/test/emqx_bridge_v1_compatibility_layer_SUITE.erl [hunk 2]
```diff
@@ -681,3 +696,91 @@ t_scenario_1(_Config) ->
...
+    {ok, #{<<"id">> := RuleId0}} =
+        emqx_bridge_v2_testlib:create_rule_and_action_http(
+            bridge_type(),
+            RuleTopic,
+            [
+                {bridge_name, BridgeName}
+                | Config
+            ],
+            #{overrides => #{enable => true}}
+        ),
```
Comment (kjellwinblad) [inline apps/emqx_bridge/test/emqx_bridge_v1_compatibility_layer_SUITE.erl:726]: Should this not fail so the user get feedback about the error (bridge does not exist)?

File: apps/emqx_rule_engine/src/emqx_rule_engine.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx_rule_engine/src/emqx_rule_engine.erl:484]: That's what I meant by "silently disabling it" when I suggested this solution. :joy_cat: 
Ok, I'll add a log here.

File: apps/emqx_bridge/test/emqx_bridge_v1_compatibility_layer_SUITE.erl (hunk 2 above)
Comment (thalesmg) [reply apps/emqx_bridge/test/emqx_bridge_v1_compatibility_layer_SUITE.erl:726]: I had originally suggested allowing the change but silently disabling it.  I'll add a log to warn the user about the inconsistency.

**Pull request merged**